### PR TITLE
Misc Changes to Schema

### PR DESCRIPTION
**Pull request by iskandarzulkarnaien**
Misc Changes to Schema

### CODE REVIEW

File: sql/create.sql [hunk 1]
```diff
@@ -1,37 +1,45 @@
...
+	tpoints integer DEFAULT 100,					/* Total accumalated reward points (upon joining: all members enjoy 100 points free) */
+    cpoints integer DEFAULT 100,                    /* Current points user has after using points for Discount */
+	rating numeric (1,2) DEFAULT 5.00,				/* Default Passenger rating */
 	CHECK (tpoints >= 0),							/* Check if tpoints is valid */
+	CHECK (cpoints >= 0),							/* Check if cpoints is valid */ 
 	CHECK (											/* Check if mstatus is valid */
-			mstatus = 'Member'
-			OR mstatus = 'Premium'
-			OR mstatus = 'VIP'
+			UPPER(mstatus) = 'MEMBER'
```
Comment (qjqqyy) [inline sql/create.sql]: same here

File: sql/create.sql [hunk 2]
```diff
@@ -1,37 +1,45 @@
...
 	name varchar(15), 								/* Nickname that is preferred */
 	gender char(1), 								/* F = Female, M = Male */
 	dob date, 										/* Date of birth */
 	addr varchar(100), 								/* Address */
 	cdate date DEFAULT current_date,				/* Date of account creation */
 	CHECK (cdate >= current_date),					/* Check if date is valid */
-	CHECK (current_date - dob >= 16)				/* Check if user is 16 y/o and above */
+	CHECK (date_part('year', age(dob)) >= 16),		/* Check if user is 16 y/o and above */
+    CHECK (                                         /* Check gender */
+            UPPER(gender) = 'M'
```
Comment (qjqqyy) [inline sql/create.sql]: if we're using the value like an enum `upper` in here confuses select statements

File: sql/create.sql [hunk 3]
```diff
@@ -1,37 +1,45 @@
...
 
 CREATE TABLE Passenger (
 	uname varchar(15) PRIMARY KEY REFERENCES Users
 		ON DELETE CASCADE,
-	mid	SERIAL NOT NULL,							/* Candidate key: Membership ID */
 	mstatus varchar(10) DEFAULT 'Member',			/* Membership status */
-	tpoints integer DEFAULT 100,					/* Accumulative reward points (upon joining: all members enjoy 100 points free) */
+	tpoints integer DEFAULT 100,					/* Total accumalated reward points (upon joining: all members enjoy 100 points free) */
+    cpoints integer DEFAULT 100,                    /* Current points user has after using points for Discount */
+	rating numeric (1,2) DEFAULT 5.00,				/* Default Passenger rating */
```
Comment (qjqqyy) [inline sql/create.sql]: we can update this (and driver ratings) with triggers right? (aggregate on every update to Transactions)

File: sql/create.sql [hunk 4]
```diff
@@ -41,155 +49,160 @@ CREATE TABLE Car (
...
+
 /*---------------------------- 		PASSENGER PART		 ----------------------------------------*/
 CREATE TABLE Reward (
 	rcode char(7) PRIMARY KEY,						/* Reward code (Eg. ABC0000) */
-	rid SERIAL NOT NULL,							/* Candidate key: Reward ID */
+	expired boolean DEFAULT false,					/* States if reward has expired */
 	CHECK (											/* D = Discount, P = Point */
-			rcode LIKE 'D%'
-		   	OR rcode LIKE 'P%'
+			UPPER(rcode) LIKE 'D%'
```
Comment (qjqqyy) [inline sql/create.sql]: same here

File: sql/create.sql [hunk 5]
```diff
@@ -41,155 +49,160 @@ CREATE TABLE Car (
...
+	FOREIGN KEY (r_redeem)
+		REFERENCES Reward (rcode),
+	FOREIGN KEY (duname, plate_num, origin, dest, ptime, pdate)
+		REFERENCES Ride (uname, plate_num, origin, dest, ptime, pdate),
+	CHECK (puname <> duname),						/* Make sures that the passenger */
+													/* and driver are not the same person */
+	CHECK (											/* Passenger ratings: Range of 0 to 5 inclusive */
 			prating <= 5
-		   	OR prating >= 0
+		   	AND prating >= 0
```
Comment (qjqqyy) [inline sql/create.sql:205]: good fix 👍

File: sql/create.sql [hunk 6]
```diff
@@ -41,155 +49,160 @@ CREATE TABLE Car (
...
-		  ),
-	CHECK (											/* Check if company is valid: System only applicable for Master/Visa transactions */
-			company = 'MASTER'
-		   	OR company = 'VISA'
-		  )
-);
 
-CREATE TABLE Transacts (
+CREATE TABLE Transactions (
+	tcode varchar(15) PRIMARY KEY,
```
Comment (qjqqyy) [inline sql/create.sql]: why not reference a Ride instead?
Doesn't a ride and passenger unambiguously identify a transaction?

File: sql/create.sql [hunk 7]
```diff
@@ -41,155 +49,160 @@ CREATE TABLE Car (
...
 		  ),
-	CHECK (											/* Driver ratings are in range of 0 to 5 inclusive */
+	CHECK (											/* Driver ratings: Range of 0 to 5 inclusive */
 			drating <= 5
-		   	OR drating >= 0
+		   	AND drating >= 0
+		  )
+);
+
+CREATE TABLE Payment (
```
Comment (qjqqyy) [inline sql/create.sql]: this (in its current state) looks like an additional column in transactions

File: sql/create.sql (hunk 3 above)
Comment (skyaxe97) [reply sql/create.sql]: Added trigger

**Pull request merged**